src/components/table/table.tsx

```diff
@@ -0,0 +1,158 @@
+import { CSSProperties, useMemo, useState, FC } from 'react';
+import styles from './table.module.scss';
+import classNames from 'classnames/bind';
+import { ArrowDownIcon, ArrowUpIcon } from '@components/icons';
+import { FixedColumn, PrimaryColumn, RowData, TableComponentProps, SortDirection } from './types';
+import { Checkbox } from '@/components';
+
+const cx = classNames.bind(styles);
+
+export const Table: FC<TableComponentProps> = ({
+  data,
+  primaryColumn,
+  fixedColumns,
+  rowActionMenu,
+  className = '',
+  selectable = false,
+  selectedRowIds = [],
+  sortingDirection = SortDirection.ASC,
+  sortingColumn = primaryColumn,
+  onChangeSorting = () => {},
+  onToggleRowSelection = () => {},
+  onToggleAllRowsSelection = () => {},
+}) => {
+  const [hoveredColumn, setHoveredColumn] = useState<string | null>(null);
+  const [hoveredRow, setHoveredRow] = useState<number | null>(null);
+
+  const sortedColumns: (PrimaryColumn | FixedColumn)[] = useMemo(() => {
+    return [{ ...primaryColumn, primary: true }, ...fixedColumns.sort((a, b) => a.order - b.order)];
```

Comment:
Can the order be handled just by the column position in the array?

Reply:
Added for faster reorder if needed, I can remove, if there will not be cases like that